samza-core/src/main/java/org/apache/samza/zk/ZkJobCoordinator.java

```diff
@@ -104,14 +116,29 @@ public synchronized void stop() {
       coordinatorListener.onJobModelExpired();
     }
 
+    metrics.isLeader.set(0);
```

Comment:
Can you please add a comment here explaining why we are resetting the gauge here?